plugins/topology/src/utils/workload-node-utils.ts

```diff
@@ -219,3 +219,89 @@ export const getKindAbbrColor = (kind: string): ResKindAbbrColor => {
   const kindColor = kindObj?.color;
   return { kindStr, kindAbbr, kindColor };
 };
+
+export enum ComputedStatus {
+  Cancelling = 'Cancelling',
+  Succeeded = 'Succeeded',
+  Failed = 'Failed',
+  Running = 'Running',
+  'In Progress' = 'In Progress',
+  FailedToStart = 'FailedToStart',
+  PipelineNotStarted = 'PipelineNotStarted',
+  Skipped = 'Skipped',
+  Cancelled = 'Cancelled',
+  Pending = 'Pending',
+  Idle = 'Idle',
+  Other = '-',
+}
+
+export enum SucceedConditionReason {
+  PipelineRunCancelled = 'StoppedRunFinally',
+  PipelineRunStopped = 'CancelledRunFinally',
+  TaskRunCancelled = 'TaskRunCancelled',
+  Cancelled = 'Cancelled',
+  PipelineRunStopping = 'PipelineRunStopping',
+  PipelineRunPending = 'PipelineRunPending',
+  TaskRunStopping = 'TaskRunStopping',
+  CreateContainerConfigError = 'CreateContainerConfigError',
+  ExceededNodeResources = 'ExceededNodeResources',
+  ExceededResourceQuota = 'ExceededResourceQuota',
+  ConditionCheckFailed = 'ConditionCheckFailed',
+}
+
+export const pipelineRunStatus = (
+  pipelineRun: any,
+): ComputedStatus | undefined => {
+  const conditions = pipelineRun?.status?.conditions ?? [];
+  if (conditions.length === 0) return undefined;
+
+  const succeedCondition = conditions.find((c: any) => c.type === 'Succeeded');
+  const cancelledCondition = conditions.find(
+    (c: any) => c.reason === 'Cancelled',
+  );
+
+  if (
+    [
+      SucceedConditionReason.PipelineRunStopped,
+      SucceedConditionReason.PipelineRunCancelled,
+    ].includes(pipelineRun.spec?.status) &&
+    !cancelledCondition
+  ) {
+    return ComputedStatus.Cancelling;
+  }
+
+  if (!succeedCondition || !succeedCondition.status) {
```

Comment:
Prefer using an optional chain expression instead, as it's more concise and easier to read.

```suggestion
  if ( !succeedCondition?.status) {
```